packages/core/src/node/test/cluster-test-worker.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+// tslint:disable:no-console
+import 'reflect-metadata';
+import { clusterRemoteMasterProcessFactory, ServerProcess } from '../cluster';
+
+const jobs: { [id: string]: (() => Promise<void>) | undefined } = {
+    restart: async () => {
+        const server = new ServerProcess(clusterRemoteMasterProcessFactory);
+        let firstRestartFailed = true;
+        try {
+            await server.restart();
+            firstRestartFailed = false;
+        } catch (e) {
+            if ((e as Error).message.indexOf('failed to restart') === -1) {
+                throw e;
+            }
+        }
+        if (firstRestartFailed) {
+            await server.restart();
+        } else {
+            throw new Error('first restart should fail');
```

Comment:
Why should the first restart fail ?

Reply:
To test that the worker is got notified about it.